### DISCUSSION TITLE
Navigate to external device by ip address dynamically

### DISCUSSION THREAD

**Discussion by tichaonax** [home-assistant/frontend#8239] [Dashboard] [answered]
Here is what I am trying to do if you can help. I have some esp8266 controllers that I have made that run custom code to control things, each has a web server running on it that I can go to from a browser and configure things like schedules, device names etc. So I can have a device that turns on a light and another that turns on  fishpond pump or water pump with its own schedule etc. I have mqtt homeassistant integration with autodiscovery enable,  my custom esp8266 devices are automatically picked up by homeassistant without doing any configuration on homeassistant side. To change settings on these devices I simply open browser and enter the ip address of the device and the web server running on the esp8266 allows me to make any changes need. Now there could be as many as 10 of these and on my network getting dynamic IP addresses. I want to be able to navigate to UI running on these devices from homeassistant.

I can turn on/off these devices from home assistant and view history from the basic lovelace UI. What I want to achieve is when I get  to the more details screen of the entity or some other way have a way to add a button or support such that when clicked it navigates to the url of the entity outside of homeassistant. This would require all this to be dynamically created so that a new device connected to the network works right away without additional configuration on homeassitant. Since the entities connect to homeassistanat via mqtt there must be  way to get the entity ip address on the fly and then navigate to its ip address as URL on some tap actin. I could not find this functionality anywhere on the forums. I really want that functionality or a way to be able to achieve it most importantly without having to configure homeassistant for each device. Thanks a lot, hope what I am saying makes sense and thanks for your work.

**Comment by spacegaier** [comment, marked as answer]
I think this is more a backend topic. If an entity has an attribute that contains a valid HTTP or HTTPS URL we will display it as a link. So if you can get the target URL / IP in the backend and add as an attribute, the frontend would show it in the more-info dialog as it does with any other attribute.
With templating you can [add custom attributes](https://www.home-assistant.io/integrations/template/#add-custom-attributes).

![image](https://user-images.githubusercontent.com/114137/105838640-a6ba5e00-5fd0-11eb-994b-7dd39d2988a4.png)

**Reply by spacegaier** [reply]
Also cross-post: https://community.home-assistant.io/t/navigate-to-ip-address-on-tap-action/272176

=> Locking discussion here, since a support topic and not a real, concrete FR.
